app/controllers/permissions_controller.rb

```diff
@@ -1,4 +1,17 @@
 class PermissionsController < ApplicationController
+  def new
+    render locals: { permission: Permission.new }
+  end
+
+  def create
+    permission = Permission.new(permission_params)
+    if permission.save
+    else
+      flash[:alert] = permission.errors.full_messages[0]
+    end
```

Comment:
if there's nothing under "if permission.save", you could change it to "unless permission.save" (and remove the "else")